components/admin_console/billing/billing_summary/index.tsx

```diff
@@ -41,12 +41,35 @@ const BillingSummary = ({isFreeTrial, daysLeftOnTrial, onUpgradeMattermostCloud}
         body = <UpgradeToProfessionalCard/>;
     } else if (isFreeTrial) {
         body = freeTrial(onUpgradeMattermostCloud, daysLeftOnTrial);
-    } else if (subscription?.last_invoice) {
+    } else if (subscription?.last_invoice && !subscription?.upcoming_invoice) {
         const invoice = subscription!.last_invoice;
-        const fullCharges = invoice.line_items.filter((item) => item.type === 'full');
+        let fullCharges = invoice.line_items.filter((item) => item.type === 'full');
         const partialCharges = invoice.line_items.filter((item) => item.type === 'partial');
+        if (!partialCharges.length && !fullCharges.length) {
+            fullCharges = invoice.line_items;
+        }
         body = (
-            lastInvoiceInfo(invoice, product, fullCharges, partialCharges)
+            <InvoiceInfo
+                invoice={invoice}
+                product={product}
+                fullCharges={fullCharges}
+                partialCharges={partialCharges}
+            />
+        );
+    } else if (subscription?.upcoming_invoice) {
+        const invoice = subscription!.upcoming_invoice;
```

Comment:
Nit: I don't think the `!` assertion is needed here because the else if already checks it. If I remove it locally my editor doesn't complain.